depthai_sdk/src/depthai_sdk/managers/pipeline_manager.py

```diff
@@ -536,6 +536,15 @@ def createEncoder(self, cameraName, encFps=30, encQuality=100):
                 raise RuntimeError("Right mono camera not initialized. Call createRightCam(res, fps) first!")
             encIn = self.nodes.monoRight.out
 
+        # set preferred profile for camera if set
+        if preferredProfile:
+            if preferredProfile == ".h264":
```

Comment:
preferredProfile should be already an instance of `dai.VideoEncoderProperties.Profile`. A correct parsing for CLI args (that are specific for demo) should be added to `ConfigManager` (which is not part of the SDK)